### DISCUSSION TITLE
0.5.4

### DISCUSSION THREAD

**Discussion by z-huang** [z-huang/music#1310] [Announcements]
- Fix YouTube API changes
- Show library artist albums first in new releases
- More actions in Android Auto
- Long click on back arrow to go to home
- More improvements and fixes

<hr />

Known issues:

- Login feature isn't working

**Comment by muelsyse-az** [comment]
LETS GOOOOOOOOO

**Reply by Ashkumar7** [reply]
Lets F***ing Go

**Reply by tronikelis** [reply]
Lets go

**Reply by greeeen-dev** [reply]
WE ARE SO BACK 🗣️🔥

**Comment by LightslicerGP** [comment]
AYAAAA!AAAA

I just switched to the fork yesterday but still, LETS GOOI

**Reply by mon-jai** [reply]
Which fork are you using?

**Reply by LightslicerGP** [reply]
https://github.com/Malopieds/InnerTune

**Reply by 7heMech** [reply]
@LightslicerGP bad news, I had switched two, my backup from the fork doesn't work here lol

**Reply by LightslicerGP** [reply]
> @LightslicerGP bad news, I had switched two, my backup from the fork doesn't work here lol

I know, I tried too and it wouldn't work

I'm sticking to the fork I guess 💀👍🏾

(It also still has more features that I like, so it only makes sense)

**Reply by Malopieds** [reply]
Please see [this](https://github.com/z-huang/InnerTune/discussions/1310#discussioncomment-10256982)

**Reply by mon-jai** [reply]
@Malopieds Any chance that some parts of the fork could be merged back upstream?

**Reply by Malopieds** [reply]
> Any chance that some parts of the fork could be merged back upstream?

Yep, most fixes will be implemented, if there are some features that I missed, feel free to open an issue on my repo !

**Comment by hasanelfalakiy** [comment]
show library artist in new release still not working

**Reply by z-huang** [reply]
Can you open or reference an issue for that?

**Reply by hasanelfalakiy** [reply]
> Can you open or reference an issue for that?

okay I'll open the issue

**Comment by f321x** [comment]
Great to see development continue🚀

**Comment by pdimu** [comment]
Even if I will still use the malopieds fork for a while. I am so proud of z-huang. Can we make a way to get backups from the fork back to the original innertune? That's the only thing that frustrates me - I can't switch back

**Reply by z-huang** [reply]
@Malopieds said he will make a migration script between his and my repo.

**Reply by Malopieds** [reply]
Actually, you just have to pass the version from 14 to 12 and bypass the hash test by putting `8db3d5731dbcc716a90427d4dde63c66` in the identity_hash table. You can use any sqlite editor to do this, or even python. If you want me to put a simple python script that does what I explained, just ping me here.

**Reply by 7heMech** [reply]
> Actually, you just have to pass the version from 14 to 12 and bypass the hash test by putting `8db3d5731dbcc716a90427d4dde63c66` in the identity_hash table. You can use any sqlite editor to do this, or even python. If you want me to put a simple python script that does what I explained, just ping me here.

Hey, when I get to my PC. I'll make a static website where you can upload your backup and it'll do that with JS. I'll post the url here

**Reply by pdimu** [reply]
> > Actually, you just have to pass the version from 14 to 12 and bypass the hash test by putting `8db3d5731dbcc716a90427d4dde63c66` in the identity_hash table. You can use any sqlite editor to do this, or even python. If you want me to put a simple python script that does what I explained, just ping me here.
> 
> 
> 
> Hey, when I get to my PC. I'll make a static website where you can upload your backup and it'll do that with JS. I'll post the url here

Made it @7heMech ?

**Reply by 7heMech** [reply]
@pdimu [this is it](https://innertune.replit.app/).

**Reply by pdimu** [reply]
omg this is so basic. i love it

**Reply by 7heMech** [reply]
@pdimu previously it used a server to modify the db, now I've updated it to be completely local, so it should be way faster.

**Reply by pdimu** [reply]
> @pdimu previously it used a server to modify the db, now I've updated it to be completely local, so it should be way faster.

You're making cool stuff in less a day. Love it

**Reply by pdimu** [reply]
And I've tried it just now with a real backup. Unfortunately on restore InnerTune crashed. I'm gonna try later I guess

**Reply by 7heMech** [reply]
@pdimu can you possibly make a PR with example backup file, because I can't seem to be able to reproduce this.

**Reply by pdimu** [reply]
A backup from MaloTune (my name for forked innertune from malopieds: Malo[pieds][Inner]Tune)? And PR on the repo of the website I guess

**Reply by Rwje** [reply]
can u please do something for importing from z huang innertune debug to urs? It crashes when I restore dubug app's backup to ur app or any other fork

**Comment by WSTxda** [comment]
This is amazing!

**Comment by houyan0111** [comment]
I'm glad to see the developers back. Will there be continuous updates in the future?

**Reply by z-huang** [reply]
I'll do my best to make updates.

**Comment by ruskcoder** [comment]
The view playlist page in account is not working, is it just me?

**Reply by z-huang** [reply]
Can you login your account?

**Comment by Grevellia123** [comment]
Thank you for updating this application

**Comment by 7heMech** [comment]


**Reply by z-huang** [reply]
F-Droid usually takes longer time to update.

**Comment by 7heMech** [comment]
If you're migrating from @Malopieds awesome fork and your backup file is not working, [this is a utility tool](https://innertune.replit.app/) I made to fix your backup file.

All credits to @Malopieds I wouldn't have known how to do this if it wasn't for his advice.

**Reply by 7heMech** [reply]
Also [this is open source](https://github.com/7heMech/InnerTune-Backup-Fix) btw :)

**Reply by Malopieds** [reply]
Great, I'll PR if I make some changes !

**Reply by 7heMech** [reply]
> Also [this is open source](https://github.com/7heMech/InnerTune-Backup-Fix) btw :)

@pdimu if you can submit an issue to the repo above that would be awesome :)

**Reply by pdimu** [reply]
Sure

**Reply by LightslicerGP** [reply]
Will the main repo get updates that the fork has? Like the wiggly bar for example?

**Reply by 7heMech** [reply]
> Will the main repo get updates that the fork has? Like the wiggly bar for example?

The @z-huang has pointed out that the main repo will have a different design from the fork, but I think some UI features might be merged, more so if you make an issue about it.

**Reply by mon-jai** [reply]
> Also [this is open source](https://github.com/7heMech/InnerTune-Backup-Fix) btw :)

Maybe consider adding a license file?

**Reply by 7heMech** [reply]
> > Also [this is open source](https://github.com/7heMech/InnerTune-Backup-Fix) btw :)
> 
> Maybe consider adding a license file?

Did, thanks.

**Reply by Rwje** [reply]
can u please do something for importing from z huang innertune debug to urs? It crashes when I restore dubug app's backup to ur app or any other fork

**Comment by Yuicmd** [comment]
Radio is not working even in fork

**Reply by z-huang** [reply]
Now fixed.
